conda/install.py

```diff
@@ -839,12 +838,21 @@ def extract(dist):
     fname = rec['files'][0]
     assert url and fname
     pkgs_dir = dirname(fname)
+
     path = fname[:-8]
     with Locked(path):
         temp_path = path + '.tmp'
         rm_rf(temp_path)
+
+        # check whether has enough space to extract a file
+        from .fetch import check_size
         with tarfile.open(fname) as t:
+            size = 0
+            for m in t.getmembers():
+                size += m.size
+            check_size(pkgs_dir, size)
```

Comment:
I agree ! check in advance will be good, but there is complexity in implementation. In fetch and extraction, we get the path of file in extract and fetch method, so it is simple to check there. However, if we check in advance, we need to figure out the destination first, and they may go to different places.

Or we can just check whether there is enough space in prefix, this is a simple solution